### PR TITLE
MCP-1677: re-enable security config

### PR DESCRIPTION
**Pull request by dimitri-amida**
# Re-enable security

### What was the problem?

Security of the API routes has been temporarily disabled because it was breaking integration tests

### How does this fix it?

- Re-enabled security
- Enhanced integration tests to use authorization
- Made JSON objects less sensitive to missing fields.

### Jira Tickets

- [MCP-1677](https://amida.atlassian.net/browse/MCP-1677)

## How to test this PR

- Go to http://localhost:8080/swagger-ui/index.html?configUrl=/v3/api-docs/swagger-config#/
- Click on the padlock next to an API.  A modal will appear. Enter a valid authorization key 

## Reminders

- Review [Pull-Requests](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests) guidelines
- [ ] If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) and
[Roadmap](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Roadmap) wiki pages
- [ ] If changing software behavior, post a link to the PR in Slack channel #benefits-virtual-regional-office

### CODE REVIEW

File: api/src/main/java/gov/va/vro/api/responses/FetchPdfResponse.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
 @Setter
 @AllArgsConstructor
 @Builder
-@ToString(includeFieldNames = true)
```
Comment (dimitri-amida) [inline api/src/main/java/gov/va/vro/api/responses/FetchPdfResponse.java:9]: That's the default

File: api/src/main/java/gov/va/vro/api/responses/HealthDataAssessmentResponse.java [hunk 2]
```diff
@@ -21,7 +21,7 @@ public class HealthDataAssessmentResponse {
   private String diagnosticCode;
 
   @Schema(description = "Medical evidence supporting assessment")
-  @JsonInclude(JsonInclude.Include.ALWAYS)
```
Comment (dimitri-amida) [inline api/src/main/java/gov/va/vro/api/responses/HealthDataAssessmentResponse.java:24]: That's the default

File: app/src/test/java/gov/va/vro/VroControllerTest.java [hunk 3]
```diff
@@ -45,26 +59,53 @@ void postHealthAssessment() {
...
         claimRepository.findByClaimSubmissionIdAndIdType("1234", "va.gov-Form526Submission");
     assertTrue(claimEntityOptional.isPresent());
   }
+
+  @Test
+  void generatePdf() throws Exception {
+    adviceWith(
+        camelContext,
+        "generate-pdf",
+        route ->
```
Comment (dimitri-amida) [inline app/src/test/java/gov/va/vro/VroControllerTest.java:89]: Replace rabbitmq endpoint with mock

File: controller/src/main/java/gov/va/vro/controller/VroController.java [hunk 4]
```diff
@@ -28,20 +27,18 @@
 @RestController
 public class VroController implements VroResource {
 
-  // https://www.baeldung.com/constructor-injection-in-spring#implicit-constructor-injection
   private final CamelEntrance camelEntrance;
   private final GeneratePdfRequestMapper generatePdfRequestMapper;
-  private final FetchPdfRequestMapper fetchPdfRequestMapper;
```
Comment (dimitri-amida) [inline controller/src/main/java/gov/va/vro/controller/VroController.java:34]: Not used

File: controller/src/main/java/gov/va/vro/controller/VroController.java [hunk 5]
```diff
@@ -28,20 +27,18 @@
...
   private final PostClaimRequestMapper postClaimRequestMapper;
 
+  private final ObjectMapper objectMapper = new ObjectMapper();
```
Comment (dimitri-amida) [inline controller/src/main/java/gov/va/vro/controller/VroController.java:34]: use the same mapper for all conversions

**Review by ianwolson — APPROVED** [review]

File: api/src/main/java/gov/va/vro/api/model/AbdBloodPressure.java [hunk 6]
```diff
@@ -1,11 +1,15 @@
 package gov.va.vro.api.model;
```
Comment (wlin2019) [inline api/src/main/java/gov/va/vro/api/model/AbdBloodPressure.java:1]: These files look like duplication of the same files in service-data-access. I guess the change here probably won't affect the others.

**Review by wlin2019 — APPROVED** [review]
LGTM

**Pull request merged**